db/cache/charts.go

```diff
@@ -1158,3 +1162,21 @@ func poolValueChart(charts *ChartData, bin binLevel, axis axisType) ([]byte, err
 	}
 	return nil, InvalidBinErr
 }
+
+func stakedCoinsChart(charts *ChartData, bin binLevel, axis axisType) ([]byte, error) {
+	switch bin {
+	case BlockBin:
+		return charts.encode(charts.Blocks.Time, accumulate(charts.Blocks.NewAtoms),
+			charts.Blocks.PoolValue)
```

Comment:
We can reduce the API response size by more that 1/3 by performing the conversion to staked percent here. That's at least 5 MB on the blocks view. You can then reconstruct the data client-side using a similar procedure to `circulationFunc` in `charts_controller`.

Reply:
I have had a looks at doing the % calculations on the backend but that implementation results to a increase in data needed to plot this chart. IMO we should still do the % calculations on the frontend.

Reply:
If we drop the pool size and circulation from the legend, and just send percents, the API response goes from 1 array of timestamps and two arrays of atoms to 1 timestamps and 1 floats. It should be about ~1/3 smaller. For blocks data that is about 5 MB. 

I'm only pointing out that the cost for the extra legend data is pretty steep. It's fine though.